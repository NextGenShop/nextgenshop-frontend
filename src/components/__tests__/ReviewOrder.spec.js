import React from "react";
import { shallow } from "enzyme";
import { Review } from "../ReviewOrder";
import mockBasketItems from "../../store/mock/MockBasketItems.json";

describe("ReviewOrder", () => {
  const defaultProps = {
    basket: {},
  };

  test("should correctly render", () => {
    const component = shallow(<Review />);
    expect(component).toMatchSnapshot();
  });

  test("should correctly render with props", () => {
    const component = shallow(<Review {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  test("should correctly render a single basket item", () => {
    const mockBasket = { items: [mockBasketItems[0]], totalPrice: 0 };
    const component = shallow(<Review {...defaultProps} basket={mockBasket} />);
    expect(component).toMatchSnapshot();
  });

  test("should correctly render multiple basket items", () => {
    const mockBasket = {
      items: [mockBasketItems[0], mockBasketItems[1]],
      totalPrice: 0,
    };
    const component = shallow(<Review {...defaultProps} basket={mockBasket} />);
    expect(component).toMatchSnapshot();
  });
});
