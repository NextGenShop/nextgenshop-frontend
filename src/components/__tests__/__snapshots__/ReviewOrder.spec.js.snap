// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewOrder should correctly render 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="h6"
  >
    Order summary
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(List))
    disablePadding={true}
  >
    <WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Total"
      />
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-total-2"
        variant="subtitle1"
      >
        £
        0
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        gutterBottom={true}
        variant="h6"
      >
        Shipping
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
      >
        John Smith
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
      >
        Gower Street, No 167, London, WCE 6AP, UK
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        gutterBottom={true}
        variant="h6"
      >
        Payment details
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card type
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Visa
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card holder
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Mr John Smith
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card number
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            xxxx-xxxx-xxxx-1234
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Expiry date
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            04/2024
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;

exports[`ReviewOrder should correctly render a single basket item 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="h6"
  >
    Order summary
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(List))
    disablePadding={true}
  >
    <WithStyles(ForwardRef(ListItem))
      key="Milk Chocolate Biscuits 9 Pack"
    >
      <WithStyles(ForwardRef(ListItemText))
        primary="Milk Chocolate Biscuits 9 Pack"
      />
      <WithStyles(ForwardRef(Typography))
        variant="body2"
      >
        £
        1.50
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Total"
      />
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-total-2"
        variant="subtitle1"
      >
        £
        0
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        gutterBottom={true}
        variant="h6"
      >
        Shipping
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
      >
        John Smith
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
      >
        Gower Street, No 167, London, WCE 6AP, UK
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        gutterBottom={true}
        variant="h6"
      >
        Payment details
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card type
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Visa
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card holder
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Mr John Smith
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card number
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            xxxx-xxxx-xxxx-1234
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Expiry date
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            04/2024
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;

exports[`ReviewOrder should correctly render multiple basket items 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="h6"
  >
    Order summary
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(List))
    disablePadding={true}
  >
    <WithStyles(ForwardRef(ListItem))
      key="Milk Chocolate Biscuits 9 Pack"
    >
      <WithStyles(ForwardRef(ListItemText))
        primary="Milk Chocolate Biscuits 9 Pack"
      />
      <WithStyles(ForwardRef(Typography))
        variant="body2"
      >
        £
        1.50
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="Cookie Dough Vanilla Ice Cream 465Ml"
    >
      <WithStyles(ForwardRef(ListItemText))
        primary="Cookie Dough Vanilla Ice Cream 465Ml"
      />
      <WithStyles(ForwardRef(Typography))
        variant="body2"
      >
        £
        4.50
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Total"
      />
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-total-2"
        variant="subtitle1"
      >
        £
        0
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        gutterBottom={true}
        variant="h6"
      >
        Shipping
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
      >
        John Smith
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
      >
        Gower Street, No 167, London, WCE 6AP, UK
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        gutterBottom={true}
        variant="h6"
      >
        Payment details
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card type
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Visa
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card holder
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Mr John Smith
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card number
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            xxxx-xxxx-xxxx-1234
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Expiry date
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            04/2024
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;

exports[`ReviewOrder should correctly render with props 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="h6"
  >
    Order summary
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(List))
    disablePadding={true}
  >
    <WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Total"
      />
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-total-2"
        variant="subtitle1"
      >
        £
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        gutterBottom={true}
        variant="h6"
      >
        Shipping
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
      >
        John Smith
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
      >
        Gower Street, No 167, London, WCE 6AP, UK
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        gutterBottom={true}
        variant="h6"
      >
        Payment details
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card type
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Visa
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card holder
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Mr John Smith
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Card number
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            xxxx-xxxx-xxxx-1234
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            Expiry date
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={6}
        >
          <WithStyles(ForwardRef(Typography))
            gutterBottom={true}
          >
            04/2024
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
